metadata/db.go

```diff
@@ -323,18 +323,15 @@ func (m *DB) GarbageCollect(ctx context.Context) (gc.Stats, error) {
 	m.dirty = 0
 
 	if len(m.dirtySS) > 0 {
-		var sl sync.Mutex
 		stats.SnapshotD = map[string]time.Duration{}
 		wg.Add(len(m.dirtySS))
 		for snapshotterName := range m.dirtySS {
 			log.G(ctx).WithField("snapshotter", snapshotterName).Debug("schedule snapshotter cleanup")
 			go func(snapshotterName string) {
 				st1 := time.Now()
 				m.cleanupSnapshotter(snapshotterName)
-
-				sl.Lock()
+				// Don't need a lock here since every goroutine is updating a unique snapshot
 				stats.SnapshotD[snapshotterName] = time.Since(st1)
```

Comment:
I think it is still data race :)

https://play.golang.org/p/ZyRjL551eHs

Reply:
@fuweid in theory yes since the goroutines will access the same map, however my reasoning was that since every goroutine will update its own key in the map only and so in practice there will be no overwriting of eachother's data. Let me know your thoughts.

Reply:
Go's race detector is not going to be OK with this. The built-in map type does not consider these sort of unlocked map writes as safe, no matter the key values.

If the map were pre-allocated with pointers, then accessing and updating would be safe, however, that is probably not more efficient. What is the performance issue you are running into today?

Reply:
@dmcgowan I haven't done any benchmark yet, don't think there would be huge performance bottlenecks.

Reply:
@gaurav1086 As other maintainers mentioned, go will detect the concurrent write, even if the key is unique. It is unsafe to modify map value without mutex. we can find other way to improve the performance, but not this pr. :)

Reply:
@fuweid totally get it. Thanks for the review. :)